packages/quick_actions/quick_actions_ios/ios/Classes/FLTQuickActionsPlugin.m

```diff
@@ -1,99 +0,0 @@
-// Copyright 2013 The Flutter Authors. All rights reserved.
-// Use of this source code is governed by a BSD-style license that can be
-// found in the LICENSE file.
-
-#import "FLTQuickActionsPlugin.h"
-#import "FLTQuickActionsPlugin_Test.h"
-#import "FLTShortcutStateManager.h"
-
-static NSString *const kChannelName = @"plugins.flutter.io/quick_actions_ios";
-
-@interface FLTQuickActionsPlugin ()
-@property(nonatomic, strong) FlutterMethodChannel *channel;
-/// The type of the shortcut item selected when launching the app.
-@property(nonatomic, strong, nullable) NSString *launchingShortcutType;
-@property(nonatomic, strong) FLTShortcutStateManager *shortcutStateManager;
-@end
-
-@implementation FLTQuickActionsPlugin
-
-+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
-  FlutterMethodChannel *channel =
-      [FlutterMethodChannel methodChannelWithName:kChannelName
-                                  binaryMessenger:[registrar messenger]];
-  FLTQuickActionsPlugin *instance =
-      [[FLTQuickActionsPlugin alloc] initWithChannel:channel
-                                shortcutStateManager:[[FLTShortcutStateManager alloc] init]];
-  [registrar addMethodCallDelegate:instance channel:channel];
-  [registrar addApplicationDelegate:instance];
-}
-
-- (instancetype)initWithChannel:(FlutterMethodChannel *)channel
-           shortcutStateManager:(FLTShortcutStateManager *)shortcutStateManager {
-  if ((self = [super init])) {
-    _channel = channel;
-    _shortcutStateManager = shortcutStateManager;
-  }
-  return self;
-}
-
-- (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result {
-  if ([call.method isEqualToString:@"setShortcutItems"]) {
-    [self.shortcutStateManager setShortcutItems:call.arguments];
-    result(nil);
-  } else if ([call.method isEqualToString:@"clearShortcutItems"]) {
-    [self.shortcutStateManager setShortcutItems:@[]];
-    result(nil);
-  } else if ([call.method isEqualToString:@"getLaunchAction"]) {
-    result(nil);
-  } else {
-    result(FlutterMethodNotImplemented);
-  }
-}
-
-- (void)dealloc {
-  [_channel setMethodCallHandler:nil];
-  _channel = nil;
```

Comment:
> Also double checked other plugins and they don't call `setMethodCallHandler:nil`

It's been a while since I dug into plugin lifetime, but IIRC it's the channel handler registration that keeps the plugin instance alive in the first place; if that's true, then this will never do anything in `dealloc` since we couldn't be here if there's still an active channel. I may be misremembering though.

Reply:
Gotcha. This would be a serious memory issue for our plugins. I created https://github.com/flutter/flutter/issues/111206 to track it.